src/SafeIam.ts

```diff
@@ -0,0 +1,86 @@
+import SafeAppSdk from '@gnosis.pm/safe-apps-sdk';
+import { bigNumberify } from 'ethers/utils';
+import { IApp, IOrganization, IRole } from './cacheServerClient/cacheServerClient.types';
+import { CacheClientNotProvidedError, ERROR_MESSAGES } from './errors';
+import { ENSNamespaceTypes, IAM } from './iam';
+import { ConnectionOptions, Transaction } from './iam/iam-base';
+
+export class SafeIam extends IAM {
+  protected _safeAddress = '';
+
+  constructor(private safeAppSdk: SafeAppSdk, iamOpts: ConnectionOptions) {
+    super(iamOpts);
+  }
+
+  protected async send(tx: Transaction) {
+    const safeTxGas = bigNumberify(await this._transactionOverrides.gasLimit || '').toNumber();
+
+    if (tx.from === this._safeAddress) {
+      await this.safeAppSdk.txs.send({
+        txs: tx.calls.map((tx) => ({ ...tx, value: tx.value || '0' })),
```

Comment:
Why the `tx.value || '0'` code? It doesn't look this this is being done in the `IAM` base class. Is it a Gnosis specific thing?

Also, is `||` the correct operator? If `tx.value == false`, then we want `value: 0`? (Maybe we do, just checking 😄 )

Reply:
Yes, unlike `ethers` signer `gnosis` signer require value parameter. `tx.value` shouldn't be false because send function is typed and accepts only `EncodedCall` which  restricts value to `string` or `undefined`